Update information on deprecated backwards naming - issue #648 (#649)
civicrm/civicrm-dev-docs: docs/extensions/index.md

## Extension Names

> Pick a unique single word for your extension's name. Note that this name is used throughout the extension's code, e.g. function names so it should only include characters that can be safely used in that way.

Reviewer: ```suggestion
Pick a unique string for your extension's name. Note that this name is used throughout the extension's code, e.g. in function names so it should only include characters that can be safely used in that way.

!!! warning
    There is currently no "canonical list" of extension names. We'd strongly recommend you check the [extensions directory](https://civicrm.org/extensions/) and the [GitLab extensions group](https://lab.civicrm.org/extensions).
```

Reviewer: Those are good questions @totten @mlutfy can you provide clarity with regards the policy used for GitLab namespaces and in-app limitations?

Reviewer 2: Minor readability nitpick: on line 16, could you add a comma beteween "names" and "so"? Then it would read "... code, e.g. in function names, so ..."

Reviewer 2: > how you edit things ↑ here like @MikeyMJCO did.

Yeah, I was wondering the same thing.

Reviewer: > > how you edit things ↑ here like @MikeyMJCO did.
> 
> Yeah, I was wondering the same thing.

In "Review"/"Files Changed" when you over over a line you'll get a + button - you can then click the little "Insert a Suggestion" button (CTRL+G)

Reviewer: ```suggestion
Pick a unique single word for your extension's name. Note that this name is used throughout the extension's code, e.g. function names, so it should only include characters that can be safely used in that way.
```

> Nb. Historically, extension names used to follow the same convention as Java package names – they look like reversed domain names. (e.g.  `com.example.myextension`), but this is no longer the recommendation.

Reviewer: ```suggestion
``` info
    Historically, extension names used to follow the same convention as Java package names – they look like reversed domain names. (e.g.  `com.example.myextension`), but this is no longer the recommendation.
```
